packages/swapper/src/swappers/RelaySwapper/endpoints.ts

```diff
@@ -0,0 +1,170 @@
+import { Err } from '@sniptt/monads/build'
+import type { InterpolationOptions } from 'node-polyglot'
+
+import type {
+  CommonTradeQuoteInput,
+  EvmTransactionRequest,
+  GetTradeRateInput,
+  GetUnsignedEvmTransactionArgs,
+  SwapErrorRight,
+  SwapperApi,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../types'
+import { TradeQuoteError } from '../../types'
+import { checkSafeTransactionStatus, isExecutableTradeQuote, makeSwapErrorRight } from '../../utils'
+import { relayChainMap } from './constant'
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { getTradeRate } from './getTradeRate/getTradeRate'
+import { relayService } from './utils/relayService'
+import type { RelayStatus } from './utils/types'
+
+export const relayApi: SwapperApi = {
+  getTradeQuote: async (
+    input: CommonTradeQuoteInput,
+    deps: SwapperDeps,
+  ): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+    if (input.sellAmountIncludingProtocolFeesCryptoBaseUnit === '0') {
+      return Err(
+        makeSwapErrorRight({
+          message: 'sell amount too low',
+          code: TradeQuoteError.SellAmountBelowMinimum,
+        }),
+      )
+    }
+
+    const tradeQuoteResult = await getTradeQuote(input, deps, relayChainMap)
+
+    return tradeQuoteResult
+  },
+  getTradeRate: async (
+    input: GetTradeRateInput,
+    deps: SwapperDeps,
+  ): Promise<Result<TradeRate[], SwapErrorRight>> => {
+    if (input.sellAmountIncludingProtocolFeesCryptoBaseUnit === '0') {
+      return Err(
+        makeSwapErrorRight({
+          message: 'sell amount too low',
+          code: TradeQuoteError.SellAmountBelowMinimum,
+        }),
+      )
+    }
+
+    const tradeRateResult = await getTradeRate(input, deps, relayChainMap)
+
+    return tradeRateResult
+  },
+  getUnsignedEvmTransaction: async ({
+    chainId,
+    from,
+    stepIndex,
+    tradeQuote,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const { to, value, data, gas, maxFeePerGas, maxPriorityFeePerGas } =
+      tradeQuote.steps[stepIndex]?.relayTransactionMetadata ?? {}
+
+    if (
+      to === undefined ||
+      value === undefined ||
+      data === undefined ||
+      gas === undefined ||
+      maxFeePerGas === undefined ||
+      maxPriorityFeePerGas === undefined
+    ) {
+      const undefinedRequiredValues = [
+        to,
+        value,
+        data,
+        gas,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+      ].filter(value => value === undefined)
+
+      throw Error('undefined required values in swap step', {
+        cause: {
+          undefinedRequiredValues,
+        },
+      })
+    }
+
+    return await Promise.resolve({
+      to,
+      from,
+      value,
+      data,
+      chainId: Number(fromChainId(chainId).chainReference),
+      maxFeePerGas,
+      maxPriorityFeePerGas,
```

Comment:
spread as these are optional for EIP1559 support e.g something like
```typescript
      gasLimit: BigNumber.max(gasLimit, estimatedGas ?? '0').toFixed(),
      ...feeData,
```

Reply:
Done in [7e3a115](https://github.com/shapeshift/web/pull/9188/commits/7e3a115c7aeb37843ca13844d6f4f73501a40f03)

Reply:
Closing this comment to not dupe discussions. See https://github.com/shapeshift/web/pull/9188#discussion_r2030278200.